src/security/core/tests/config.c

```diff
@@ -591,6 +589,21 @@ CU_Test(ddssec_config, other_prop, .init = ddsrt_init, .fini = ddsrt_fini)
     "</Domain>";
 
   CU_ASSERT_FATAL((qos = dds_create_qos()) != NULL);
+  dds_qset_prop(qos, DDS_SEC_PROP_AUTH_LIBRARY_PATH, WRAPPERLIB_PATH("dds_security_authentication_wrapper"));
+  dds_qset_prop(qos, DDS_SEC_PROP_AUTH_LIBRARY_INIT, "init_test_authentication_all_ok");
+  dds_qset_prop(qos, DDS_SEC_PROP_AUTH_LIBRARY_FINALIZE, "finalize_test_authentication_all_ok");
+  dds_qset_prop(qos, DDS_SEC_PROP_CRYPTO_LIBRARY_PATH, WRAPPERLIB_PATH("dds_security_cryptography_wrapper"));
+  dds_qset_prop(qos, DDS_SEC_PROP_CRYPTO_LIBRARY_INIT, "init_test_cryptography_all_ok");
+  dds_qset_prop(qos, DDS_SEC_PROP_CRYPTO_LIBRARY_FINALIZE, "finalize_test_cryptography_all_ok");
+  dds_qset_prop(qos, DDS_SEC_PROP_ACCESS_LIBRARY_PATH, WRAPPERLIB_PATH("dds_security_access_control_wrapper"));
+  dds_qset_prop(qos, DDS_SEC_PROP_ACCESS_LIBRARY_INIT, "init_test_access_control_all_ok");
+  dds_qset_prop(qos, DDS_SEC_PROP_ACCESS_LIBRARY_FINALIZE, "finalize_test_access_control_all_ok");
+  dds_qset_prop(qos, "dds.sec.auth.identity_ca", TEST_IDENTITY_CA_CERTIFICATE_DUMMY);
+  dds_qset_prop(qos, "dds.sec.auth.private_key", TEST_IDENTITY_PRIVATE_KEY_DUMMY);
+  dds_qset_prop(qos, "dds.sec.auth.identity_certificate", TEST_IDENTITY_CERTIFICATE_DUMMY);
+  dds_qset_prop(qos, "dds.sec.access.permissions_ca", "file:Permissions_CA.pem");
+  dds_qset_prop(qos, "dds.sec.access.governance", "file:Governance.p7s");
+  dds_qset_prop(qos, "dds.sec.access.permissions", "file:Permissions.p7s");
```

Comment:
@clalancette, the idea behind this test is that security settings from the (xml) configuration are used only when no QoS properties exist that have a name that starts with `dds.sec.`. If any property with this prefix is set, then the security settings from the xml configuration are ignored and all required settings should be set in the QoS. The test case checks this behavior. 

The comment above the test case is not very clear. The terms 'config' and 'config security elements' are confusing and should be something like '_Cyclone XML configuration_' and '_security element in the Cyclone XML configuration_'. 

I think the check 'any security properties in the QoS' should also include the vendor specific prefix that is introduced by this PR.

Reply:
Hm, OK.  The reason I made this change at all is that this test was failing to load the security modules, which ended up in a crash.  I honestly don't understand the module loading mechanism at present, so I wasn't sure what was going on here.  I'll take another look at this test and see if I can figure out the right thing to do.

Reply:
This turned out to be a bug in ddsi_plist.c; see 38f1e56 .  I've now restored this code to the original (plus the small changes needed for the new property).